filter_plugins/oo_filters.py

```diff
@@ -377,6 +377,10 @@ def oo_parse_certificate_names(certificates, data_dir, internal_hostnames):
             else:
                 certificate['names'] = []
 
+            # We will always look for named certificates in data_dir regardless of their original path.
+            certificate['certfile'] = os.path.join(data_dir, os.path.basename(certificate['certfile']))
+            certificate['keyfile'] = os.path.join(data_dir, os.path.basename(certificate['keyfile']))
```

Comment:
I think it might still be a good idea...  having the temporary location may be good anyway, it might be better to parse the certs prior to copying them over to the host.  How difficult would it be to do the parsing in the filter and then the fact could store just the remote location to the certs?

I'm just thinking about additional runs and not wanting to have the certs required to be present for every run.

What do you think about having the cert fact be additive and when certs are specified they will overwrite certs if present on the remote or add to the certs if not.

We'd still need a way to remove certs if wanted though.